packages/material-ui/src/Modal/Modal.js

```diff
@@ -208,6 +208,7 @@ const Modal = React.forwardRef(function Modal(props, ref) {
 
   const inlineStyle = styles(theme || { zIndex });
   const childProps = {};
+  // FixMe: Always apply document role. Revisit once React Flare is released
```

Comment:
What do you mean by "always"? I mean, how is it different from what we do now?

Reply:
Poor wording on my part. Wanted to express that we should fix that it always applies the role.

Reply:
Oh, I was already expecting the role to be always applied. I guess you have found a case where it doesn't.

Reply:
Nah.
1. the role itself is incorrect most of the time
2. It overwrites roles of the child